src/WebServer_I2C_Scanner.ino

```diff
@@ -5,72 +5,135 @@
 // ********************************************************************************
 // Web Interface I2C scanner
 // ********************************************************************************
-void handle_i2cscanner_json() {
-  checkRAM(F("handle_i2cscanner"));
-
-  if (!isLoggedIn()) { return; }
-  navMenuIndex = MENU_INDEX_TOOLS;
-  TXBuffer.startJsonStream();
-  json_init();
-  json_open(true);
 
+int scanI2CbusForDevices_json( // Utility function for scanning the I2C bus for valid devices, with JSON output
+        int8_t muxAddr
+      , int8_t channel
+      , int nDevices
+#ifdef FEATURE_I2CMULTIPLEXER
+      , String &excludeDevices
```

Comment:
Why use a String here?
A `std::vector<bool>` is a lot cheaper to use here.
A vector of bool is a special kind of vector, which uses a packed representation.
So of length 128 does only use 128 bits, which is 16 bytes (+ the few bytes the vector container uses)

Searching through a String of numbers separated with a `|` roughly 1000 times is quite inefficient.

Reply:
Must be my lack on C++ knowledge, I'm reading up on this `<vector>` stuff.

Reply:
Just the very basics:
- You have to set the length using `reserve` yourself, or else you will get out of range issues when addressing it.
- Address the values simply by `variable[x]`.

For example:

```c++
typedef std::vector<bool> i2c_addresses_t;

i2c_addresses_t i2c_address;
i2c_address.resize(128);
for (int i = 0; i < 128; ++i) {
  i2c_address[i] = false;
}
```